### ISSUE TITLE
configure path to caching JSON data

### ISSUE THREAD

**Issue by paulz1** [BurntSushi/nflgame#163]
At first thank for your application :-)

I just try to launch you example in README.md (about rushing yards) but for 2015 season and for some first weeks.

If I launch : games = nflgame.games(2015,week=[1,2,3,4]) it works great.
But if I launch : games = nflgame.games(2015,week=[1,2,3,4,5])
I get a lot of 
"Could not cache JSON data. Please make '/usr/lib/python2.6/site-packages/nflgame/gamecenter-json' writable."

The things that I don't understand :
a) the week 5 is over... so why? It seems that this problme raise when we try access the weeks that has not been played yet... but the Week 5 has?
b) I'm not a fan of virtualenv... or setting 777 to system directories.
So may be we could just set where the JSON cache should be saved?

For info :
I'm in Scientific Linux 6.7 ( = CentOS 6.7 we could say).
And nflgame was installed used pip.

**Comment by BurntSushi** [comment]
If you installed nflgame and nflgame didn't come with week 5 data and nflgame could never cache that data then it will always fetch it from NFL.com because it was never cached. You are quite literally running into that exact problem: nflgame knows that week 5 is over, so it's trying to cache the data, but it can't.

> I'm not a fan of virtualenv

Sorry, but that's the standard way to install Python packages these days. I don't see any good reason to work around it.

**Comment by BurntSushi** [comment]
With that said, configuring where JSON data is cached may be a worthwhile endeavor. The data is growing every year, so the Python package directory may not be the best place to put it. Other Python packages that have a similar problem are `nltk`, but I've always found their approach annoying. Basically, you have to explicitly download the nltk data to directory, and if you don't, some of nltk's functionality won't work. The problem with that approach for us is that nflgame will work just fine without a cache (because, you know, it really is a cache). Which means it would be very easy for users to completely neglect downloading data, and we'd all end up hitting NFL.com way too much. So... some more thought needs to go into this.

**Comment by paulz1** [comment]
Ok, thank you very much for explication.
I didn't realize that nflgame has cached games inside it when we install it (I thought it's just check is the week is over and then download data from NFL.com).

So I close my ticket thank you very much for help. I will search some hint to give the rights for the JSON-cache directory.

Nevertheless, I'm agreed with your reflections. I don not think that save the data inside package directory is a good idea.

**Comment by BurntSushi** [comment]
No, this is still a good ticket! I just don't consider it essential right now, so I'm not sure when I'll get to it.

**Comment by BurntSushi** [comment]
If you use virtualenv on a directory you own, then you don't need to do anything. Cached game data will be saved inside that virtualenv which should be writable.

**Comment by dangsteven** [comment]
I'm running into the same issue. I've installed nflgame into the virtualenv but it's still throwing the same error.

**Comment by BurntSushi** [comment]
@dangsteven Allow me to clarify: you must create a virtualenv in a directory that you have permission to write to, then activate the virtualenv, then install nflgame and then use nflgame while the virtualenv is activated.

Please include the full sequence of steps you took, including the commands you ran and their outputs. If you don't include this information, then it will be very hard for anyone to help you.
